react-frontend/src/pages/CreateListingPage.js

```diff
@@ -7,21 +8,62 @@ import TextAreaInput from '../components/TextAreaInput/TextAreaInput'
 import NumberInput from '../components/NumberInput/NumberInput'
 import Button from '../components/Button/Button'
 import Popup from '../components/Popup/Popup'
+import Alert from '../components/Alert/Alert'
 import { createListing } from '../api'
+import { AuthContext } from '../../context/AuthContext'
 import './styles/CreateListingPage.css'
 
 const CreateListingPage = () => {
+  const { isAuthenticated } = useContext(AuthContext)
+  const navigate = useNavigate()
   const [title, setTitle] = useState('')
-  const [category, setCategory] = useState('')
+  const [currency, setCurrency] = useState('')
   const [images, setImages] = useState([])
   const [description, setDescription] = useState('')
   const [price, setPrice] = useState('')
   const [isPopupOpen, setIsPopupOpen] = useState(false)
   const [isSuccess, setIsSuccess] = useState(false)
   const [errorMessage, setErrorMessage] = useState('')
+  const [showAuthAlert, setShowAuthAlert] = useState(false)
+  const [showValidationAlert, setShowValidationAlert] = useState(false)
+
+  const currencyOptions = [
+    { value: 'UAH', label: 'UAH' },
+    { value: 'USD', label: 'USD' },
+    { value: 'EUR', label: 'EUR' }
+  ]
+
+  useEffect(() => {
+    if (!isAuthenticated) {
+      setShowAuthAlert(true)
+      const timer = setTimeout(() => {
+        navigate('/login')
+      }, 3000)
+      return () => clearTimeout(timer)
+    }
+  }, [isAuthenticated, navigate])
+
+  const validateForm = () => {
+    return (
+      title.trim() !== '' &&
+      currency !== '' &&
+      description.trim() !== '' &&
+      price.trim() !== '' &&
+      images.length > 0
+    )
+  }
 
   const handleSubmit = (e) => {
     e.preventDefault()
+    if (!isAuthenticated) {
+      setShowAuthAlert(true)
+      return
+    }
+    if (!validateForm()) {
```

Comment:
It would be helpful if we validate each field separately.

Reply:
Added validation in [1745875](https://github.com/YaroslavKSE/AuctionWeb/pull/3/commits/174587596ab873a6ae4b2fbf2caf2a3cd19a0569)